ElrondNetwork/elrond-go
# En 7369 stop node after offline sync

- added the gracefully close channel in storage header resolver so the import process can safely shutdown the node when completed

Manual testing procedures:
- provide a db directory containing data in continuous epoch.
- move this directory into a new directory, might be `import-db`, place the directory near the `config` directory.
- the process can be run either with a partially build `db` directory or without a DB.
- change the `DestinationShardAsObserver` option in prefs.toml to match the containing DB directories and desired shard.
- use the same config files that have generated the initial import-db data, otherwise the node can not advance
- run the node with `./node -use-log-view -log-level *:INFO -log-save -import-db ./import-db`.
- the process should iterate on each and every block in advancing in every existing epoch
- the process stops when it hits an epoch that it has a missing data (this will usually occur after traversing the whole import-db directory)

Note: please check that the node has at least 3 peer connections, otherwise the sync and consensus mechanism are paused.
